### PR TITLE
Allow users recover which class it is given obfuscated class name

### PR DESCRIPTION
**Pull request by fzyzcjy**
Closes https://github.com/flutter/flutter/issues/126330

### CODE REVIEW

**Comment by fzyzcjy** [comment]
> We expect to start accepting PRs again the week of May 10th.

Today is May 10th :) I can wait for a few days for the PR

File: src/deployment/obfuscate.md [hunk 1]
```diff
@@ -116,6 +116,23 @@ use the following steps to make it human readable:
    For more information on the `symbolize` command,
    run `flutter symbolize -h`.
 
+## Reading an obfuscated name
+
+To understand a name created by an obfuscated app,
+use the following steps to make it human readable:
+
+1. At app build time, save the name obfuscation map
+   using `--extra-gen-snapshot-options=--save-obfuscation-map=/<your-directory>`.
```
Comment (domesticmouse) [inline src/deployment/obfuscate.md:125]: I'm curious why this line has a directory, yet the next command has a path. Does the tool generate it's own file(s) in the named target directory? Worth adding that clarification, I believe.

Comment (fzyzcjy) [reply src/deployment/obfuscate.md]: My bad, it is (always) a path, fixed

**Review by domesticmouse — APPROVED** [review]

File: src/deployment/obfuscate.md [hunk 2]
```diff
@@ -116,6 +116,23 @@ use the following steps to make it human readable:
    For more information on the `symbolize` command,
    run `flutter symbolize -h`.
 
+## Reading an obfuscated name
+
+To understand a name created by an obfuscated app,
+use the following steps to make it human readable:
+
+1. At app build time, save the name obfuscation map
+   using `--extra-gen-snapshot-options=--save-obfuscation-map=/<your-path>`.
```
Comment (atsansone) [inline src/deployment/obfuscate.md:125]: **issue:** Use parallel structure.
```suggestion
1. To save the name obfuscation map at app build time,
   use `--extra-gen-snapshot-options=--save-obfuscation-map=/<your-path>`.
```

File: src/deployment/obfuscate.md [hunk 3]
```diff
@@ -116,6 +116,23 @@ use the following steps to make it human readable:
...
+   For example:
+
+   ```terminal
+   $ flutter build apk --obfuscate --split-debug-info=/<project-name>/<directory> --extra-gen-snapshot-options=--save-obfuscation-map=/<your-path>
+   ```
+
+1. Use the generated obfuscation map to recover the name.
+   The obfuscation map is a simple JSON array where
+   each obfuscated name is preceded by its original name.
```
Comment (atsansone) [inline src/deployment/obfuscate.md:134]: **issue:** Condition, then instruction per [GSG](https://developers.google.com/style/sentence-structure).

```suggestion
1.  To recover the name, use the generated obfuscation map.
   The obfuscation map is a small JSON array where
   each obfuscated name is preceded by its original name.
```

**question:** Is this an array of JSON objects? That would give you pairs, which would be a more basic explanation.

**Review by atsansone — COMMENTED** [review]
The JSON array structure needs explanation plus a few GSG changes.

File: src/deployment/obfuscate.md (hunk 3 above)
Comment (fzyzcjy) [reply src/deployment/obfuscate.md]: Yes it is pairs, I will update that

**Comment by fzyzcjy** [comment]
Sure! I will update soon

**Comment by fzyzcjy** [comment]
(updated)

**Comment by atsansone** [comment]
@fzyzcjy : Just one question and a clean up.

**Comment by fzyzcjy** [comment]
@atsansone I did update that (answer the question and change words to "pairs"), just did not mark it resolved so you can see it ;)

**Review by sfshaza2 — APPROVED** [review]
lgtm. This looks good, thanks!

**Pull request merged**